docs/design/interface/pseudocode/write_resource_data_to_raw.py

```diff
@@ -0,0 +1,120 @@
+# ruff: noqa
+def write_resource_data_to_raw(data_path, path, resource_properties):
+    """Writes the raw data file into the resource's raw data folder.
+
+    Copy the file from `data_path` over into the resource location given by
+    `path`. This will compress the file and use a timestamped, unique file
+    name to store it as a backup. See the
+    [design](https://sprout.seedcase-project.org/docs/design/) docs for an
+    explanation of this file. Use `path_resource_raw()` to provide the
+    correct `path` location. Copies and compresses the file, and outputs the
+    path object of the created file.
+
+        Args:
+            data_path: Path to a raw data file that you want Sprout to store
+                into the data package.
+            path: Path to the raw resources folder. Use `path_resources_raw()`
+                to help provide the correct path.
+            resource_properties: The properties object for the specific resource.
+                Use `read_properties()` to read the properties for the resource
+                and `get_properties()` to get the correct resource properties.
+
+        Returns:
+            The path to the written compressed raw resource file.
+
+        Examples:
+            ```{python}
+            #| eval: false
+            ```
+    """
+    check_is_dir(path)
+    check_is_file(data_path)
+    check_is_supported_format(data_path)
+    check_data_basics(data_path, resource_properties)
+    check_data_constraints(data_path, resource_properties)
+
+    raw_resource_path = Path(path / create_raw_file_name(data_path))
+    return write_compressed_file(data_path, raw_resource_path)
+
+
+def create_raw_file_name(data_path: Path) -> Error | None:
```

Comment:
```suggestion
def create_raw_file_name(data_path: Path) -> str:
```
This could just return a string, right? No need to make it a `Path`